engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/WorldPreGenerationScreen.java

```diff
@@ -15,31 +15,258 @@
  */
 package org.terasology.rendering.nui.layers.mainMenu;
 
+import com.google.common.collect.Lists;
 import org.terasology.assets.ResourceUrn;
+import org.terasology.config.Config;
+import org.terasology.context.Context;
+import org.terasology.engine.module.ModuleManager;
+import org.terasology.math.TeraMath;
+import org.terasology.module.ModuleEnvironment;
+import org.terasology.registry.In;
+import org.terasology.rendering.assets.texture.Texture;
+import org.terasology.rendering.assets.texture.TextureData;
 import org.terasology.rendering.nui.CoreScreenLayer;
+import org.terasology.rendering.nui.NUIManager;
 import org.terasology.rendering.nui.WidgetUtil;
-import org.terasology.world.generation.World;
+import org.terasology.rendering.nui.databinding.Binding;
+import org.terasology.rendering.nui.layers.mainMenu.preview.FacetLayerPreview;
+import org.terasology.rendering.nui.layers.mainMenu.preview.PreviewGenerator;
+import org.terasology.rendering.nui.widgets.UIDropdownScrollable;
+import org.terasology.rendering.nui.widgets.UIImage;
+import org.terasology.rendering.nui.widgets.UISlider;
+import org.terasology.rendering.world.WorldSetupWrapper;
+import org.terasology.utilities.Assets;
+import org.terasology.world.generator.UnresolvedWorldGeneratorException;
+import org.terasology.world.generator.WorldGenerator;
+import org.terasology.world.generator.internal.WorldGeneratorManager;
+import org.terasology.world.generator.plugin.TempWorldGeneratorPluginLibrary;
+import org.terasology.world.generator.plugin.WorldGeneratorPluginLibrary;
+import org.terasology.world.zones.Zone;
 
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.stream.Collectors;
+
+/**
+ * This class let's the user preview different worlds added in the
+ * {@link UniverseSetupScreen}. Each world is still configurable and it's seed
+ * can be changed by the re-roll button. Note that each world has a unique seed.
+ */
 public class WorldPreGenerationScreen extends CoreScreenLayer {
 
     public static final ResourceUrn ASSET_URI = new ResourceUrn("engine:worldPreGenerationScreen");
+    @In
+    private ModuleManager moduleManager;
+    @In
+    private Config config;
+    private ModuleEnvironment environment;
+    private WorldGenerator worldGenerator;
+    private Texture texture;
+    private UIImage previewImage;
+    private Context context;
+    private PreviewGenerator previewGen;
+    private List<WorldSetupWrapper> worldList;
+    private String selectedWorld;
+    private List<String> worldNames;
+    private int seedNumber = 0;
+    private float zoomValue = 2f;
+
+    /**
+     * A function called before the screen comes to the forefront to setup the environment
+     * and extract necessary objects from the new Context.
+     * @param subContext As there is a new environment created in the {@link UniverseSetupScreen}, it's passed to
```

Comment:
@param subContext The new environment created in UniverseSetupScreen.

Reply:
(you can then add why you pass a new one instead of the old one, but you don't write stories inside the javadoc).